CONNECTION_POOL.md

```diff
@@ -0,0 +1,149 @@
+
+
+## Overview
+
+The lifecycle of a connection pool of the `node-nuodb` driver to a **NuoDB** database entails:
+
+- Initialization of the connection `Pool`
+- Request connections on the `pool`
+- Use the connections
+- Release each connection
+- Close the `pool`
+
+For simplicity the examples to follow will use `try/catch` semantics, although you `Promise` semantics is also permitted. 
+
+The following is a general example requesting 2 connections on the connection pool. You would in turn manage each connection as a [single connection](./SINGLE_CONNECTION.md#execute-sql).
+
+```js
+const connectionConfig = {
+    database: `test`,
+    password: "dba",
+    user: "dba",
+    port: "48004",
+    schema: "USER"
+};
+
+const poolConfig = {
+    minAvailable: 10,
+    connectionConfig,
+    maxAge: 2000,
+    checkTime: 10000,
+    maxLimit: 12,
+    connectionRetryLimit: 5,
+};
+
+async function() {
+    let pool = null;
+
+    try {
+        pool = new Pool(poolConfig);
+        const conn1 = await pool.requestConnection();
+        const conn2 = await pool.requestConnection();
+        // use each connection
+
+        await pool.releaseConnection(conn1);        
+        await pool.releaseConnection(conn2);        
+
+    } catch (err) {/* handle error */}
+
+    finally {
+        try {
+            if (pool != null) {
+                await pool.closePool()
+            }
+        } catch (err) {/* handle error */}
+    }
+}
+```
+
+
+## Initialize Connection Pool
+
+To initialize a connection pool, you must provide a pool configuration. Please review the shape of the [pool configuration](#pool-configuration) to customize for specific needs. Note, you must specify the [connection configuration](./SINGLE_CONNECTION.md#connection-properties) as for a single connection. After a pool is created the user must initialize it using the `init()` method. This will populate the pool. The pool is unavailable until this completes.
+
+```js
+const poolConfig = {
+    minAvailable: 10,
+    connectionConfig,
+    maxAge: 2000,
+    checkTime: 10000,
+    maxLimit: 12,
+    connectionRetryLimit: 5,
+};
+
+async function() {
+    const pool = new Pool(poolConfig);
+    await pool.init()
+}
+```
+
+## Request Connections
+
+Once the pool has been successfully initialized it is ready for use. There are now free connections that can be requested for use. You can see the pool connections via `pool.all_connections` and `pool.free_connections`.
+
+```js
+const conn1 = await Pool.requestConnection();
+```
+
+You can request multiple connections this way. Each connection can then be used as a [single connection](./SINGLE_CONNECTION.md#execute-sql) would.
+
+## Release Connections
+
+Once the user has finished using a connection, it can be returned to the pool. You must pass each connection in turn desired to release. Don't forget to manage each single connection's lifecycle properly prior to requesting its release.
```

Comment:
"You must pass each connection in turn desired to release"
not sure what this means.
"You must return each connection individually"?